remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java

```diff
@@ -144,26 +156,117 @@ public View onCreateView(LayoutInflater inflater,
+            instanceof FirebaseRemoteControllerSyncer;
+    if (isNetworkBasedSync) {
+      expandSharingOptionsButton.setVisibility(View.VISIBLE);
+      ((FirebaseRemoteControllerSyncer) Remixer.getInstance().getSynchronizationMechanism())
+          .addSharingStatusListener(this);
+      shareDrawer.init();
+    }
     super.onResume();
   }
 
   @Override
   public void onDetach() {
     super.onDetach();
   }
+
+  @Override
+  public void updateSharingStatus(boolean sharing) {
+    sharedStatusButton.setVisibility(sharing ? View.VISIBLE : View.GONE);
+  }
+
+  private class ShareDrawerOnClickListener implements View.OnClickListener {
+
+    @Override
+    public void onClick(View view) {
+      isShowingDrawer = !isShowingDrawer;
+      if (isShowingDrawer) {
+        expandSharingOptionsButton.setImageResource(R.drawable.ic_expand_less_black_24dp);
+        expandShareDrawer();
+      } else {
+        expandSharingOptionsButton.setImageResource(R.drawable.ic_expand_more_black_24dp);
+        collapseShareDrawer();
+      }
+    }
+  }
+
+  private void collapseShareDrawer() {
+    final int initialHeight = shareDrawer.getMeasuredHeight();
+    shareDrawer.setVisibility(View.VISIBLE);
+    Animation a = new Animation() {
+      @Override
+      protected void applyTransformation(float interpolatedTime, Transformation t) {
+        if (interpolatedTime == 1) {
+          shareDrawer.setVisibility(View.GONE);
+        } else {
+          shareDrawer.getLayoutParams().height =
+              initialHeight - (int) (initialHeight * interpolatedTime);
+          shareDrawer.requestLayout();
+        }
+      }
+
+      @Override
+      public boolean willChangeBounds() {
+        return true;
+      }
+    };
+    // 225 ms is a good time for elements entering the screen.
+    // https://material.io/guidelines/motion/duration-easing.html#duration-easing-common-durations
+    a.setDuration(225);
+    shareDrawer.startAnimation(a);
+  }
+
+
+  private void expandShareDrawer() {
+    shareDrawer.measure(
+        LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
+    final int targetHeight = shareDrawer.getMeasuredHeight();
+    shareDrawer.setVisibility(View.VISIBLE);
+    // Workaround for API < 21
+    shareDrawer.getLayoutParams().height = 1;
+    Animation a = new Animation() {
+      @Override
+      protected void applyTransformation(float interpolatedTime, Transformation t) {
+        shareDrawer.getLayoutParams().height =
+            interpolatedTime == 1
+                ? LinearLayout.LayoutParams.WRAP_CONTENT
+                : (int) (targetHeight * interpolatedTime);
+        shareDrawer.requestLayout();
+      }
+
+      @Override
+      public boolean willChangeBounds() {
+        return true;
+      }
+    };
+    // 195ms is a good time for elements leaving the screen.
+    // https://material.io/guidelines/motion/duration-easing.html#duration-easing-common-durations
+    a.setDuration(195);
```

Comment:
I think it would be clearer if there were variables for these animation durations.